### PR TITLE
Improve incremental tasks documentation

### PR DESCRIPTION
**Pull request by wolfs**
There was some additional feedback on #8940.

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/work/InputChanges.java [hunk 1]
```diff
@@ -28,6 +28,9 @@
  * An incremental work action is one that accepts a single {@link InputChanges} parameter.
  * The work action can then query what changed for an input parameter since the last execution to only process the changes.
  *
+ * The following example shows a task which reverses the text in each of its input files.
+ * It demonstrates how to use {@link InputChanges} to only process the changed files.
+ *
```
Comment (pledbrook) [inline subprojects/core-api/src/main/java/org/gradle/work/InputChanges.java]: :+1:

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -27,14 +27,14 @@ Switch your build to use Gradle @version@ by updating your wrapper properties:
 
 With Gradle, it's very simple to implement a task that is skipped when all of its inputs and outputs are up to date (see [Incremental Builds](userguide/more_about_tasks.html#sec:up_to_date_checks)).
 However, there are times when only a few input files have changed since the last execution, and you'd like to avoid reprocessing all of the unchanged inputs.
-Tasks only reprocessing out-of-date input files are called _incremental tasks_.
-Up to Gradle 5.4, for implementing an incremental task you would use [`IncrementalTaskInputs`](dsl/org.gradle.api.tasks.incremental.IncrementalTaskInputs.html).
+When a task has an input that can represent multiple files and the task only processes those files that are out of date, it's called an incremental task.
```
Comment (pledbrook) [inline subprojects/docs/src/docs/release/notes.md]: Did you want to remove the italics?

Comment (wolfs) [reply subprojects/docs/src/docs/release/notes.md]: I didn't, good catch! I put them back.

**Comment by wolfs** [comment]
@pledbrook Thank you for your review! I merged the changes.

**Pull request merged**